### PR TITLE
Add documentation for `pallet_assets`

### LINKED ISSUE #106: Documentation for `pallet-assets`
Currently, we have `pallet-assets` configured, but it would be nice to also document it.

### PR DESCRIPTION
**Pull request by 0xGeorgii**
Fixes #106 

- [x] Documentation (it is all about the documentation)

### CODE REVIEW

**Comment by 0xGeorgii** [comment]
hey team. I have drafted this PR to ensure I follow the correct way. I am not sure if #26 is about this PR but sounds like can be related, is that correct? 
`palette assets` was introduced many years ago which means that it has been included in `polkadot-SDK` from the very beginning, so what version I should refer to in the `Branch/Release` section?

Am I right that in the `Dispatchables` section I need to overview all `pub` functions?

**Comment by ozgunozerk** [comment]
Hi @0xGeorgii , glad to see you here 🥳 

> hey team. I have drafted this PR to ensure I follow the correct way. I am not sure if https://github.com/OpenZeppelin/polkadot-runtime-template/issues/26 is about this PR but sounds like can be related, is that correct?

Yes, this partially addresses issue #26, (for the docs). We also need to configure `pallet_assets` for our runtime, afaik, it is not there yet. 



If you want, you can try to do that yourself as well (adding `pallet_assets` to our runtime), or we can gladly help you with that.

> so what version I should refer to in the Branch/Release section?

We should use the `release-polkadot-v1.3.0` branch for now, as we are following that version for all the other pallets.

> Am I right that in the Dispatchables section I need to overview all pub functions?

No, we only need to capture the dispatchables (the ones with `call` macro on top of them). All `pub` functions may confuse the target audience for these docs, and they are out of scope.

Thanks again for your contributions in advance!

File: docs/modules/ROOT/pages/pallets/assets.adoc [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+[source,rust]
+----
+pub fn destroy_approvals(
+    origin: OriginFor<T>,
+    id: T::AssetIdParameter,
+) -> DispatchResult
+----
+Clears all approvals associated with a given asset.
+
+NOTE: TODO: consider T::RemoveItemsLimit usage.
```
Comment (ozgunozerk) [inline docs/modules/ROOT/pages/pallets/assets.adoc:127]: should we resolve this TODO?

**Comment by ozgunozerk** [comment]
Wow! Thanks a lot @0xGeorgii 

Did not know `pallet_assets` had 30 dispatchables 😱 

I believe 2 tasks remain for this PR:
- [ ] configuring `pallet_assets` for our Runtime
- [ ] we now have a dedicated `glossary` page. It would be better to move these glossary items to there, and give links to that glossary page item for the first occurrences of the words in here.

You already have done a great deal, please let us know if you want us to help for the remaining two items, or for anything else :)

**Comment by 0xGeorgii** [comment]
@ozgunozerk thank you for the feedback. I want to review dispatchables' documentation again and then move terms to the glossary. regarding the pallet configuration, I probably need some advice. Will dm you :)

File: docs/modules/ROOT/pages/pallets/assets.adoc (hunk 1 above)
Comment (0xGeorgii) [reply docs/modules/ROOT/pages/pallets/assets.adoc:127]: I will review it, thanks

**Comment by ozgunozerk** [comment]
Pallet configuration is done in another PR, this PR can solely focus on documentation :)

**Review by ozgunozerk — APPROVED** [review]

File: docs/modules/ROOT/pages/pallets/assets.adoc [hunk 2]
```diff
@@ -0,0 +1,733 @@
...
+
+**Params:**
+
+* `origin: OriginFor<T>` -- caller, must be root.
+* `id: T::AssetIdParameter` -- the identifier of the asset for which transactions are being blocked.
+* `who: AccountIdLookupOf<T>` -- the account that is being blocked from transacting the asset.
+
+== Important Mentions and FAQ’s
+
+== More Reading
```
Comment (KitHat) [inline docs/modules/ROOT/pages/pallets/assets.adoc:733]: Could you please remove sections that do not have any content?

File: docs/modules/ROOT/pages/pallets/assets.adoc [hunk 3]
```diff
@@ -0,0 +1,733 @@
...
+
+== Pallet specific terms
+
+** `Asset ID` - A unique identifier for an asset type.
+** `Asset Owner` - The account that owns an asset.
+** `Admin` - The account that has administrative privileges over an asset, such as: `minting`, `burning`, `freezing`, and `thawing`.
+** `Sufficiency` - The amount of an asset that is required to be held in an account in order to perform certain operations, such as: `transfers`, `freezing`, and `thawing`.
+
+== Config
+
```
Comment (KitHat) [inline docs/modules/ROOT/pages/pallets/assets.adoc:21]: Could you please describe the config values?

**Review by KitHat — COMMENTED** [review]
In general, the content of the documentation is very good, thank you!

However, for PR to get merged in my opinion, it is missing some parts. In addition to other comments, this PR needs two more things:

* A link to this docpage in `nav.adoc`.
* We try to add info about at least possible errors from extrinsics, and if applicable about events emitted from the extrinsic.

If you could add these changes, it would be perfect. If you have no opportunity to do it, then we will push some changes to you PR next week if you don't mind.

**Comment by 0xGeorgii** [comment]
> * We try to add info about at least possible errors from extrinsics, and if applicable about events emitted from the extrinsic.

it is added to the `nav.adoc`

regarding the quote above can you please elaborate more on what should be done? do I need to add also documentation about events that these functions emit, or enumerate events defined in the palette, or both?

@KitHat

File: docs/modules/ROOT/pages/pallets/assets.adoc (hunk 3 above)
Comment (0xGeorgii) [reply docs/modules/ROOT/pages/pallets/assets.adoc:21]: do you mean to add documentation for the types defined here?

https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/assets/src/lib.rs#L231

Comment (KitHat) [reply docs/modules/ROOT/pages/pallets/assets.adoc:21]: Yes.
Our goal is too keep the documentation in one place, so there would be no need for developers to go somewhere else for any explanations

**Comment by KitHat** [comment]
@0xGeorgii 
Regarding the errors and events I can explain the motivation to add them.
Official Polkadot documentation rarely mentions which errors may be thrown from an extrinsic and almost never explains what do they mean. As a result, frontend developers have to go through the Rust code of Substrate pallets and look where this error comes from and why does it happen which is a heavy burden for them. Describing these errors will make the documentation a great relief for the frontend developers who use Substrate chains as a backend (or a part of it).

With the events it is not that useful as the errors. Some teams use the events to build their regular backends and track the changes onchain, that's why I am not pushing for logging the events that are happening in the extrinsic.

TL;DR:
Errors that are thrown from extrinsics are really useful for the developers, for new devs (they are the target audience of this template) they are crucial. So we try to add them for every extrinsic.
Events are not that critical, they are useful for the bigger teams. They are useful, but not crucial at all.
I hope that I have resolved your questions :)

**Review by KitHat — APPROVED** [review]
@0xGeorgii Looks good to me!
Big thanks to you for adding them!

**Comment by 0xGeorgii** [comment]
@ozgunozerk  @KitHat I am not sure about the PR merging policy, so please do it on my behalf  😇
